sdks/java/extensions/jackson/src/main/java/org/apache/beam/sdk/extensions/jackson/AsJsons.java

```diff
@@ -56,6 +63,35 @@ private AsJsons(Class<? extends InputT> outputClass) {
     return newTransform;
   }
 
+  /**
+   * Returns a new {@link AsJsonsWithFailures} transform that catches exceptions raised while
+   * writing JSON elements, passing the raised exception instance and the input element being
+   * processed through the given {@code exceptionHandler} and emitting the result to a failure
+   * collection.
+   *
+   * <p>Example usage:
+   *
+   * <pre>{@code
+   * WithFailures.Result<PCollection<String>, KV<MyPojo, Map<String, String>>> result =
+   *     pojos.apply(
+   *         AsJsons.of(MyPojo.class)
+   *             .withFailures(new WithFailures.ExceptionAsMapHandler<MyPojo>() {}));
+   *
+   * PCollection<String> output = result.output(); // valid json elements
+   * PCollection<KV<MyPojo, Map<String, String>>> failures = result.failures();
+   * }</pre>
+   */
+  @Experimental(Experimental.Kind.WITH_EXCEPTIONS)
+  public <FailureT> AsJsonsWithFailures<FailureT> withFailures(
```

Comment:
I'm not sure that `withFailures` is a good name in this case and I couldn't find out if any agreement about that exists. So, any advices are welcomed.

Reply:
My bias would be towards following the API in `MapElements` as closely as possible, which would make this method `exceptionsVia` and would suggest that we should also have `exceptionsInto` and `exceptionsVia(ProcessFunction)` to allow users to define exception processing inline. I would love to see this factored out somehow so it's easier to add exception handling to different classes with a stable API and without repeating all this boilerplate, but I haven't found a suitable alternative.

Reply:
And perhaps these JSON cases are constrained enough that we could provide a default exceptionHandler so that the user doesn't need to specify one. Perhaps it would catch only the exceptions we expect to be raised (`JsonProcessingException` for AsJsons and `IOException` for ParseJsons), reraising an other exceptions, and return a KV that contains the input element along with a string containing the exception message.

Reply:
Agree to use the same API names for all transforms. Perhaps, it would make sense to add an interface for that?

Also, do we need to have separate method `exceptionsInto` only for lambdas?

And finally, yes, I also was thinking to have default exception handler. So, I think it  will be a good idea to add this, thanks.

Reply:
> Agree to use the same API names for all transforms. Perhaps, it would make sense to add an interface for that?

Yes, if it doesn't end cluttering the API too much. We could potentially put this at `WithFailures.ExceptionsVia` and it would have to be templated on `<InputT, FailureT, TransformT>` so we would end up with something like:

    implements WithFailures.ExceptionsVia<String, FailureT, AsJsonsWithFailures<FailureT>>

> Also, do we need to have separate method `exceptionsInto` only for lambdas?

Yes, I think we need a single `exceptionsInto` method. If the handler is passed as a lambda, Beam likely won't be able to infer a coder for the failure output collection without explicitly giving type information via `exceptionsInto`.

> And finally, yes, I also was thinking to have default exception handler. So, I think it will be a good idea to add this, thanks.

👍